src/lib/git-exec.js

```diff
@@ -0,0 +1,42 @@
+const exec = require('./exec-promise');
+const pluralize = require('pluralize');
+const log = require('./log');
+
+const GIT = 'git';    // Git command path
+
+/**
+ * Returns the working tree status in a string (files to commit),
+ * or empty if the working tree is clean.
+ */
+module.exports.status = async () => {
+  return (await exec(log.LEVEL_ERROR, GIT, 'status', '--porcelain')).trim();
+};
+
+/**
+ * Sets up to date the local git repository fetching from
+ * the upstream repository (but without auto-merge).
+ */
+module.exports.fetch = () => {
+  return exec(log.LEVEL_ERROR, GIT, 'fetch');
+};
```

Comment:
I think this should be merged into the `checkUpstream` function because if you forget to call it then the `checkUpstream` won't work as expected. This abstracts the implementation away from the calling function.

Reply:
Merged.